tools/add_header

```diff
@@ -0,0 +1,151 @@
+
+    # Get username and mail from git
+    cmd = 'git config --global user.name'
+    status, username = commands.getstatusoutput(cmd)
+    username = username.strip()
+    cmd = 'git config --global user.email'
+    status, email = commands.getstatusoutput(cmd)
+    email = email.strip()
+    email = '<%s>' % email
+
+    for MyFile in arguments.MyFiles:
+        # Query log history
+        cmd = '''git log --reverse --date=short --format='%s'  %s ''' % (format, MyFile)
+        status, output = commands.getstatusoutput(cmd)
+
+        # parse git log output
+        authors = collections.OrderedDict()
+        min, max = now.year, now.year
+        for line in output.split('\n'):
+            # remove leading and trailing whitespace
+            line = line.strip()
+            if line == '':
+                continue
+
+            # split the line into words
+            author, mail, date = line.split(',')
+            year = int(date.split('-')[0])
+
+            if int(min) > int(year):
+                min = year
+
+            if author not in authors:
+                authors[author] = {'name': author,
+                                   'mail': mail,
+                                   'max': year,
+                                   'min': year}
+            else:
+                if year > authors[author]['max']:
+                    authors[author]['max'] = year
+                if year < authors[author]['min']:
+                    authors[author]['min'] = year
+
+            if authors[author]['name'] == username:
+                authors[author]['max'] = now.year
+
+        if username not in authors:
+            authors[username] = {'name': username,
+                                 'mail': email,
+                                 'max': max,
+                                 'min': now.year}
+
+        header = ''
+        if arguments.script:
+            header = '#!/usr/bin/env python\n'
+
+        if int(min) == int(max):
+            header += '# Copyright %s CERN for the benefit of the ATLAS collaboration.\n' % max
+        else:
+            header += '# Copyright %s-%s CERN for the benefit of the ATLAS collaboration.\n' % (min, max)
+
+        header += '''#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+'''
+        for author in authors:
+            if int(authors[author]['min']) == int(authors[author]['max']):
+                header += '# - %(name)s, %(mail)s, %(min)s\n' % authors[author]
```

Comment:
i propose the RFC-822 style with brackets for emails

# - %(name)s <%(mail)s>, %(min)s\n

Reply:
The list of mails is already filled with the brackets. So its fine.

Reply:
Line 51

Reply:
right, but the comma after the name needs to go

Reply:
The comma after the name is historically present in almost all files.
Example: lib/rucio/common/utils.py

Reply:
yep, but it should go ;-)

Reply:
let it go